From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Haris Okanovic <haris.okanovic@ni.com>
Date: Mon, 25 Feb 2019 17:31:50 -0600
Subject: [PATCH] Revert "global_init: ignore CURL_GLOBAL_SSL's absense"

This reverts commit d661b0afb571a68ec11f3dd856aba304c80780ba.

Restore CURL_GLOBAL_SSL flag in NI-cURL to maintain
backward compatibility.

Upstream-Status: Inappropriate [deprecated behavior]

Updated for curl 7.72.

Signed-off-by: Richard Tollerton <rich.tollerton@ni.com>

diff --git a/docs/libcurl/curl_global_init.3 b/docs/libcurl/curl_global_init.3
index d3b3dc96f..4e9a07822 100644
--- a/docs/libcurl/curl_global_init.3
+++ b/docs/libcurl/curl_global_init.3
@@ -62,8 +62,8 @@ Initialize everything possible. This sets all known bits except
 \fBCURL_GLOBAL_ACK_EINTR\fP.
 
 .IP CURL_GLOBAL_SSL
-(This flag's presence or absence serves no meaning since 7.57.0. The
-description below is for older libcurl versions.)
+(This flag's presence or absence serves no meaning since 7.57.0 in
+mainline cURL. It's still observed by NI-cURL.)
 
 Initialize SSL.
 
diff --git a/include/curl/curl.h b/include/curl/curl.h
index 490b83297..0621cf14b 100644
--- a/include/curl/curl.h
+++ b/include/curl/curl.h
@@ -2680,7 +2680,7 @@ typedef enum {
   CURLCLOSEPOLICY_LAST /* last, never use this */
 } curl_closepolicy;
 
-#define CURL_GLOBAL_SSL (1<<0) /* no purpose since since 7.57.0 */
+#define CURL_GLOBAL_SSL (1<<0) /* Still observed by NI-cURL; no purpose since since 7.57.0 in mainline cURL */
 #define CURL_GLOBAL_WIN32 (1<<1)
 #define CURL_GLOBAL_ALL (CURL_GLOBAL_SSL|CURL_GLOBAL_WIN32)
 #define CURL_GLOBAL_NOTHING 0
diff --git a/lib/easy.c b/lib/easy.c
index 3b210f077..9fd1b2559 100644
--- a/lib/easy.c
+++ b/lib/easy.c
@@ -165,7 +165,9 @@ static CURLcode global_init(long flags, bool memoryfuncs)
 #endif
   }
 
-  if(!Curl_ssl_init()) {
+  /* XXX CURL_GLOBAL_SSL was removed from mainline cURL,
+   * keeping for backward compatibility */
+  if((flags & CURL_GLOBAL_SSL) && !Curl_ssl_init()) {
     DEBUGF(fprintf(stderr, "Error: Curl_ssl_init failed\n"));
     goto fail;
   }
@@ -287,6 +289,9 @@ void curl_global_cleanup(void)
     return;
   }
 
+  /* XXX CURL_GLOBAL_SSL was removed from mainline cURL,
+   * keeping for backward compatibility */
+  if(init_flags & CURL_GLOBAL_SSL)
   Curl_ssl_cleanup();
   Curl_resolver_global_cleanup();
 
-- 
2.28.0

